.github/scripts/ci/find_broken_wikilinks.py

```diff
@@ -0,0 +1,173 @@
+import os
+import sys
+
+def red(s):
+    return f"\x1b[31m{s}\x1b[0m"
+
+def green(s):
+    return f"\x1b[32m{s}\x1b[0m"
+
+def yellow(s):
+    return f"\x1b[33m{s}\x1b[0m"
+
+def blue(s):
+    return f"\x1b[34m{s}\x1b[0m"
+
+redirects = {}
+with open("wiki/redirect.yaml", 'r', encoding='utf-8') as file:
+    content = file.read()
+    for linenumber, line in enumerate(content.split('\n'), start=1):
+        split = line.split('"')
+        try:
+            redirects[split[1]] = (split[3], linenumber)
+        except Exception:
+            pass
+
+def child(path):
+    return path[path.find('/', 1) + 1:]
+
+def directory(filename):
+    return filename[filename.find('/') + 1:filename.rfind('/')]
+
+note = ""
+def check_redirect(link):
+    global note
+    link = link.lower()
+    try:
+        redirect = redirects[link]
+    except KeyError:
+        return False
+    if not os.path.exists(f"wiki/{redirect[0]}"):
+        note = f"{blue('Note:')} Broken redirect (redirect.yaml:{redirect[1]}: {link} --> {redirect[0]})"
+        return False
+    return True
+
+def check_link(directory, link):
+    global note
+    note = ""
+    if link.startswith("/wiki/"):
+        # absolute wikilink
+        if os.path.exists(link[1:]):
+            return True
+        else:
+            # may have a redirect
+            return check_redirect(child(link))
+    elif not (link.startswith("http") or link.startswith("mailto:") or link.startswith("irc:")):
+        # relative wikilink
+        if os.path.exists(f"wiki/{directory}/{link}"):
+            return True
+        else:
+            # may have a redirect
+            return check_redirect(f"{directory}/{link}")
+    else:
+        # some other link; don't care
+        return True
+
+def is_in_comment(s, start, end):
+    return s.rfind("<!--", 0, start) != -1 and s.find("-->", end, -1)
+
+def find_link(s, index=0):
```

Comment:
would be nice to have a short docstring on what it does

Reply:
if you mean `# Finds one link in a string` then i don't think that's necessary.. ?